### PR TITLE
feat(dashboard): Implement email step editor & small preview

### PR DESCRIPTION
**Pull request by desiprisg**
### What changed? Why was the change needed?

* Enable adding an email step
* Add the editor and preview tabs on the configure email template page
* Add the email editor and subject input on the editor tab

### Screenshots

Screenshot 2024-12-04 at 11 35 03 AM
Screenshot 2024-12-05 at 11 45 59 AM

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/dashboard/.env.example [hunk 1]
```diff
@@ -8,3 +8,4 @@ VITE_NOVU_APP_ID=
 VITE_INTERCOM_APP_ID=
 VITE_GTM=
 VITE_SELF_HOSTED=
+VITE_EMAIL_DELAY_DIGEST_ENABLED=
```
Comment (BiswaViraj) [inline apps/dashboard/.env.example]: are we not using the launchdarkly for FF in new dashboard?

Comment (BiswaViraj) [reply apps/dashboard/.env.example]: ok i see, we will be using launchdarkly
here is the [PR](https://github.com/novuhq/novu/pull/7133/files) to add that.
You will need to update it after that PR merges

File: apps/dashboard/src/components/workflow-editor/add-step-menu.tsx [hunk 2]
```diff
@@ -73,6 +75,7 @@ export const AddStepMenu = ({
   onMenuItemClick: (stepType: StepTypeEnum) => void;
 }) => {
   const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const areNewStepsEnabled = useFeatureFlag(FeatureFlagsKeysEnum.IS_ND_DELAY_DIGEST_EMAIL_ENABLED);
```
Comment (SokratisVidros) [inline apps/dashboard/src/components/workflow-editor/add-step-menu.tsx:78]: Lol, the feature flag name is a bit off, but it's OK.

File: apps/dashboard/src/components/workflow-editor/steps/component-utils.tsx [hunk 3]
```diff
@@ -23,6 +25,12 @@ export const getComponentByType = ({ component }: { component?: UiComponentEnum
     case UiComponentEnum.URL_TEXT_BOX: {
       return <InAppRedirect />;
     }
+    case UiComponentEnum.MAILY: {
```
Comment (SokratisVidros) [inline apps/dashboard/src/components/workflow-editor/steps/component-utils.tsx:28]: @desiprisg consider aligning the naming on this one. All other components are based on the step type field, not on the implementation of the built-in ui or the underlying library. 

```suggestion
    case UiComponentEnum.EMAIL_BODY: {
```

File: apps/dashboard/src/components/workflow-editor/steps/email/configure-email-step-preview.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import { useStep } from '@/components/workflow-editor/steps/step-provider';
+import { usePreviewStep } from '@/hooks';
+import { EmailPreviewHeader } from '@/components/workflow-editor/steps/email/email-preview';
+import { Separator } from '@/components/primitives/separator';
+import { Skeleton } from '@/components/primitives/skeleton';
+import { ChannelTypeEnum } from '@novu/shared';
+import { cn } from '@/utils/ui';
+
+type SmallEmailPreviewProps = HTMLAttributes<HTMLDivElement>;
+const SmallEmailPreview = (props: SmallEmailPreviewProps) => {
```
Comment (SokratisVidros) [inline apps/dashboard/src/components/workflow-editor/steps/email/configure-email-step-preview.tsx]: If not mistaken we call this Mini preview.

```suggestion
const MiniEmailPreview = (props: SmallEmailPreviewProps) => {
```

File: apps/dashboard/src/components/workflow-editor/steps/email/email-editor-preview.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+    return () => clearTimeout(timeout);
+  }, [editorValue]);
+
+  return (
+    <EmailTabsSection>
+      <div className="relative flex flex-col gap-3">
+        <div className="flex items-center gap-2.5 text-sm font-medium">
+          <Notification5Fill className="size-3" />
+          In-app template editor
```
Comment (SokratisVidros) [inline apps/dashboard/src/components/workflow-editor/steps/email/email-editor-preview.tsx:60]:
```suggestion
          Email template editor
```

File: apps/dashboard/src/components/workflow-editor/steps/email/maily.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+import { cn } from '@/utils/ui';
+import { Editor } from '@maily-to/core';
+import type { Editor as TiptapEditor } from '@tiptap/core';
+import { HTMLAttributes, useMemo, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+
+const bodyKey = 'emailEditor';
+
+type MailyProps = HTMLAttributes<HTMLDivElement>;
+export const Maily = (props: MailyProps) => {
```
Comment (SokratisVidros) [inline apps/dashboard/src/components/workflow-editor/steps/email/maily.tsx:13]:
```suggestion
export const Editor = (props: EditorProps) => {
```

File: apps/dashboard/src/components/workflow-editor/steps/email/maily.tsx [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+import { parseStepVariablesToLiquidVariables } from '@/utils/parseStepVariablesToLiquidVariables';
+import { cn } from '@/utils/ui';
+import { Editor } from '@maily-to/core';
+import type { Editor as TiptapEditor } from '@tiptap/core';
+import { HTMLAttributes, useMemo, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+
+const bodyKey = 'emailEditor';
+
+type MailyProps = HTMLAttributes<HTMLDivElement>;
```
Comment (SokratisVidros) [inline apps/dashboard/src/components/workflow-editor/steps/email/maily.tsx:12]:
```suggestion
type EditorProps = HTMLAttributes<HTMLDivElement>;
```

**Review by SokratisVidros — APPROVED** [review]

File: apps/dashboard/src/components/workflow-editor/add-step-menu.tsx (hunk 2 above)
Comment (desiprisg) [reply apps/dashboard/src/components/workflow-editor/add-step-menu.tsx:78]: this was already on ld

File: apps/dashboard/src/components/workflow-editor/steps/email/email-editor-preview.tsx (hunk 5 above)
Comment (desiprisg) [reply apps/dashboard/src/components/workflow-editor/steps/email/email-editor-preview.tsx:60]: not the current ticket

**Pull request merged**